src/tokenizer/lindera.rs

```diff
@@ -0,0 +1,79 @@
+use std::borrow::Cow;
+
+use lindera::tokenizer::Tokenizer as LinderaTokenizer;
+use lindera_core::core::viterbi::Mode;
+use once_cell::sync::Lazy;
+
+use crate::{Token, TokenKind};
+use crate::processors::ProcessedText;
+use super::{Tokenizer, TokenStream};
+
+
+#[derive(Debug, Default)]
+pub struct Lindera;
+
+impl Tokenizer for Lindera {
+    fn tokenize<'a>(&self, s: &'a ProcessedText<'a>) -> TokenStream<'a> {
+        let mut tokenizer = LinderaTokenizer::new(Mode::Normal, "");
+        let tokenized = tokenizer.tokenize(&s.processed);
+        TokenStream {
+            inner: Box::new(tokenized.into_iter().scan(0, move |_, lindera_token| {
+                let char_start = 0;
+                let char_end = lindera_token.text.len();
+                Some(Token {
+                    kind: TokenKind::Word,
+                    word: Cow::Borrowed(lindera_token.text),
+                    char_index: 0,
+                    byte_start: char_start,
+                    byte_end: char_end,
+                })
+            }))
```

Comment:
```suggestion
            inner: Box::new(tokenized.into_iter().scan((0, 0), move |(char_index, byte_index), lindera_token| {
                let char_count = lindera_token.text.chars().count();
                let char_start = *char_index;
                *char_index += char_count;

                let byte_count = lindera_token.text.len();
                let byte_start = *byte_index;
                *byte_index += byte_count;
                
                Some(Token {
                    kind: TokenKind::Word,
                    word: Cow::Borrowed(lindera_token.text),
                    char_index: char_start,
                    byte_start,
                    byte_end: byte_index,
                })
            }))
```

this should work. 😄

Reply:
@ManyTheFish seems it's correct. Thanks for guide 🙏